common/src/main/java/com/wynntils/utils/type/TimedValue.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils.type;
+
+import java.util.concurrent.TimeUnit;
+
+public class TimedValue<T> {
```

Comment:
This class should have a bit of a different API, to make it easier/clearer to use.

This class should behave like a provider class, with the duration/unit settings being kept as a final value. The value itself should be changeable, and should have it's expiry reset when the value changes. The get method itself should behave like an optional, aka, it should throw, if the value is expired. That way, we the developer has to explicitly check for `isExpired`.

Reply:
Perhaps it would also be beneficial to have a `reset` method, which clears the current value, and throws in get/return false in `isExpired`.